### PR TITLE
(11363) Add support for option to manage attachment on either vpn_profile/vpn_user

### PR DESCRIPTION
**Pull request by zhiqiangjin**
Solution:
 - Add a flag called "manage_user_attachment" into aviatrix_vpn_profile;
 - Add a flag called "manage_user_attachment" into aviatrix_vpn_user;
 - Add a parameter called "profiles" (list of profile name) into aviatrix_vpn_user;

### CODE REVIEW

File: goaviatrix/vpn_user.go [hunk 1]
```diff
@@ -134,3 +145,36 @@ func (c *Client) DeleteVPNUser(vpnUser *VPNUser) error {
 	}
 	return nil
 }
+
+func (c *Client) GetProfileForUser(vpnUser *VPNUser) (string, error) {
```
Comment (arvindsree) [inline goaviatrix/vpn_user.go]: lets wait for 6.0 API for refresh

File: goaviatrix/vpn_user.go [hunk 2]
```diff
@@ -36,10 +37,21 @@ type VPNUserInfo struct {
 	VpnUser VPNUser `json:"vpn_user"`
 }
 
+type GetVpnUsersResp struct {
+	Return  bool    `json:"return"`
+	Results VPNUser `json:"results"`
+	Reason  string  `json:"reason"`
+}
+
+type EditVPNUser struct {
```
Comment (arvindsree) [inline goaviatrix/vpn_user.go]: What is this structure for?

Comment (zhiqiangjin) [reply goaviatrix/vpn_user.go]: It was for previous implementation. Deleted it.

File: aviatrix/resource_aviatrix_vpn_user.go [hunk 3]
```diff
@@ -48,7 +49,18 @@ func resourceAviatrixVPNUser() *schema.Resource {
...
 				Default:     "",
 				ForceNew:    true,
-				Description: "This is the name of the SAML endpoint to which the user is to be associated.",
+				Description: "This is the name of the SAML endpoint to which the user will be associated.",
+			},
+			"profiles": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "List of profiles for user to attach to.",
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_vpn_user.go:58]: Should we give default as empty list?

File: aviatrix/resource_aviatrix_vpn_user.go [hunk 4]
```diff
@@ -129,6 +258,20 @@ func resourceAviatrixVPNUserDelete(d *schema.ResourceData, meta interface{}) err
...
 	log.Printf("[INFO] Deleting Aviatrix VPNUser: %#v", vpnUser)
 
+	if _, ok := d.GetOk("profiles"); ok {
+		log.Printf("[INFO] Found profiles: %#v", d.Get("profiles"))
+		for _, profileName := range d.Get("profiles").([]interface{}) {
+			profile := &goaviatrix.Profile{
+				Name: profileName.(string),
+			}
+			profile.UserList = append(profile.UserList, vpnUser.UserName)
+			err := client.DetachUsers(profile)
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_vpn_user.go]: I think this is automatically done in the controller

**Review by arvindsree — APPROVED** [review]

File: aviatrix/resource_aviatrix_vpn_user.go (hunk 4 above)
Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_vpn_user.go]: Removed it.

File: aviatrix/resource_aviatrix_vpn_user.go (hunk 3 above)
Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_vpn_user.go:58]: Optional is OK, I think.

**Pull request merged**